### PR TITLE
ORC-709: FIX Boolean to StringGroup schema evolution

### PR DESCRIPTION
**Pull request by pgaref**
### What changes were proposed in this pull request?
Special ConvertTreeReader for Boolean using StringGroupFromAnyIntegerTreeReader for String/Char/Varchar types for 1.6 branch

### Why are the changes needed?
Properly handle Boolean to String/Char/Varchar conversions

### How was this patch tested?
TestSchemaEvolution.testBooleanToStringEvolution

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java [hunk 1]
```diff
@@ -1744,12 +1744,65 @@ public void nextVector(ColumnVector previousVector,
...
+        throw new IllegalArgumentException("No conversion of type " +
+            readerType.getCategory() + " to self needed");
+      }
+      return new AnyIntegerFromAnyIntegerTreeReader(columnId, fileType, readerType,
+          context);
+
+    case FLOAT:
+    case DOUBLE:
+      return new DoubleFromAnyIntegerTreeReader(columnId, fileType,
+          context);
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java:1771]: Shall we make this one liner like the master branch?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except one minor style comment).
Feel free to merge this after revising that, @pgaref .
Thank you!

File: java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java (hunk 1 above)
Comment (pgaref) [reply java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java]: Yup, fixed now

**Comment by pgaref** [comment]
Thanks for the review @dongjoon-hyun ! This is now merged to 1.6 branch

**Pull request merged**